panic: index out of range when running JSONBench q3 against CSUP in vector runtime

## tl;dr

```
$ SUPER_VAM=1 super -c "
SELECT
    commit.collection AS event,
    EXTRACT(HOUR FROM time(time_us*1000)) as hour_of_day,
    count() AS count
FROM 'bluesky.csup'
WHERE kind = 'commit'
  AND commit.operation = 'create'
  AND commit.collection in ['app.bsky.feed.post', 'app.bsky.feed.repost', 'app.bsky.feed.like']
GROUP BY event, hour_of_day
ORDER BY hour_of_day, event;"

panic: runtime error: index out of range [20208] with length 20208

goroutine 142 [running]:
github.com/brimdata/super/vector.NewView({0xf5256e8, 0xc0008ac000}, {0xc00288c0c0, 0x3, 0x4})
	/Users/phil/work/super/vector/view.go:36 +0x945
github.com/brimdata/super/runtime/vam/expr.(*PredicateWalk).evalForList(0xc0025ac6c0, {0xf5256e8, 0xc0008ac000}, {0xf525808, 0xc003602000}, {0xc01b602000, 0x9c41, 0xd5b0f54?}, {0xc01b09a000, 0x4ef0, ...})
	/Users/phil/work/super/runtime/vam/expr/logic.go:302 +0x36a
github.com/brimdata/super/runtime/vam/expr.(*PredicateWalk).Eval(0xc0025ac6c0, {0xc000cd3ae0?, 0x10?, 0x2000?})
	/Users/phil/work/super/runtime/vam/expr/logic.go:263 +0x1dc
github.com/brimdata/super/runtime/vam/expr.(*In).eval(0xc004f15650, {0xc0027b6140?, 0xc000cd3b60?, 0xf525568?})
	/Users/phil/work/super/runtime/vam/expr/logic.go:232 +0xe5
github.com/brimdata/super/vector.Apply(0x80?, 0x2?, {0xc0027b6140?, 0xc0009f7180?, 0xc0027b6080?})
	/Users/phil/work/super/vector/apply.go:17 +0xa3
github.com/brimdata/super/vector.Apply(0x1, 0xc000cd3c98, {0xc0027b6080?, 0xc0006d8460?, 0xec20560?})
	/Users/phil/work/super/vector/apply.go:23 +0x10e
github.com/brimdata/super/runtime/vam/expr.(*In).Eval(0xc004f15650, {0xf525568, 0xc008ce4d00})
	/Users/phil/work/super/runtime/vam/expr/logic.go:221 +0xe5
github.com/brimdata/super/runtime/vam/expr.(*And).Eval(0xc004f15680, {0xf525568, 0xc008ce4d00})
	/Users/phil/work/super/runtime/vam/expr/logic.go:63 +0x6b
github.com/brimdata/super/runtime/vam/op.(*Filter).Pull(0xc004f156b0, 0x0)
	/Users/phil/work/super/runtime/vam/op/filter.go:25 +0x65
github.com/brimdata/super/runtime/vam/op/aggregate.(*Aggregate).Pull(0xc0004878c0, 0xf0?)
	/Users/phil/work/super/runtime/vam/op/aggregate/aggregate.go:58 +0x155
github.com/brimdata/super/runtime/vam/op.(*combineParent).run(0xc003c0b500)
	/Users/phil/work/super/runtime/vam/op/combine.go:110 +0x3c
created by github.com/brimdata/super/runtime/vam/op.(*Combine).Pull.func1 in goroutine 131
	/Users/phil/work/super/runtime/vam/op/combine.go:42 +0x35
```

## Details

Repro is with super commit 4f29a87 and a SuperSQL equivalent of the [JSONBench q3 query](https://github.com/ClickHouse/JSONBench/blob/a94914caedd64996ccc79eb3033df45c6691c867/clickhouse/queries.sql#L3).

To download/prepare the test data:

```
$ wget https://clickhouse-public-datasets.s3.amazonaws.com/bluesky/file_0001.json.gz

$ super -version
Version: v1.18.0-348-g4f29a870

$ super -f csup -o bluesky.csup file_0001.json.gz 
```

When the query is run outside of vector runtime, it does complete successfully.

```
$ super -c "
SELECT
    commit.collection AS event,
    EXTRACT(HOUR FROM time(time_us*1000)) as hour_of_day,
    count() AS count
FROM 'bluesky.csup'
WHERE kind = 'commit'
  AND commit.operation = 'create'
  AND commit.collection in ['app.bsky.feed.post', 'app.bsky.feed.repost', 'app.bsky.feed.like']
GROUP BY event, hour_of_day
ORDER BY hour_of_day, event;"

{event:"app.bsky.feed.like",hour_of_day:16,count:444523(uint64)}
{event:"app.bsky.feed.post",hour_of_day:16,count:86812(uint64)}
{event:"app.bsky.feed.repost",hour_of_day:16,count:56993(uint64)}
```

--- a/runtime/vam/expr/logic.go
+++ b/runtime/vam/expr/logic.go
@@ -285,10 +285,11 @@ func (p *PredicateWalk) evalForList(lhs, rhs vector.Any, offsets, index []uint32
 	out := vector.NewBoolEmpty(lhs.Len(), nil)
 	var lhsIndex, rhsIndex []uint32
 	for j := range lhs.Len() {
+		idx := j
 		if index != nil {
-			j = index[j]
+			idx = index[j]
 		}
-		start, end := offsets[j], offsets[j+1]
+		start, end := offsets[idx], offsets[idx+1]
 		if start == end {
 			continue
 		}

